pkg/resources/data.go

```diff
@@ -686,7 +686,7 @@ func GetCSIMigrationFeatureGates(cluster *kubermaticv1.Cluster, version *semverl
 		// The following feature gates are always enabled when the
 		// 'externalCloudProvider' feature is activated.
 		if cluster.Spec.Features[kubermaticv1.ClusterFeatureExternalCloudProvider] {
-			featureFlags = append(featureFlags, "CSIMigration=true", "ExpandCSIVolumes=true")
+			featureFlags = append(featureFlags, "CSIMigration=true")
```

Comment:
I think the CSIMigration feature flag is gone in 1.27, at least I don't see it in https://github.com/kubernetes/kubernetes/blob/release-1.27/pkg/features/kube_features.go

Reply:
CSIMigrationAWS and CSIMigrationAzureDisk are gone as well

Reply:
Once we reach 1.27 for userclusters, the CCM/CSI migration should be completed already, so this code should not have any effect. It seems all e2e tests are functioning fine, so if it's okay with you, I'd like to avoid adding more if statements to the already uber complex code that pieces together the feature gates :/

Reply:
> the CCM/CSI migration should be completed already

I don't think this assumption is correct. Where do we enforce this? It appears that - at least looking at our version configuration - the only providers enforcing CCM migrations at a certain version are OpenStack and vSphere. If I have a long running AWS cluster, are we 100% confident it's going to be migrated before being upgraded to 1.27?

Reply:
>  the only providers enforcing CCM migrations at a certain version are OpenStack and vSphere. If I have a long running AWS cluster, are we 100% confident it's going to be migrated before being upgraded to 1.27?

Exactly this. We only enforce CCM/CSI migration for OpenStack and vSphere. You might have AWS and Azure clusters running Kubernetes 1.27 and in-tree cloud provider, that yet need to migrate to CCM/CSI. That's why we need to pay attention to those flags, but also because we'll support CCM/CSI migration for additional providers in future.

Reply:
I don't know what you imagine this code to do now.

Reply:
@xrstf This CSIMigration feature gate needs to go away completely. It is completely removed in 1.27 and enabled by default starting with 1.17, so we don't need to set it at all.

From 1.17 branch: https://github.com/kubernetes/kubernetes/blob/1d9e1e6d56cbd5c9d04ba25cccf184a973056a79/pkg/features/kube_features.go#L597

From 1.27 branch (it's not there): https://github.com/kubernetes/kubernetes/blob/release-1.27/pkg/features/kube_features.go